### ISSUE TITLE
Warn/check if any 2FA provider is active before enforcing 2FA

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/server#12267]
From https://github.com/nextcloud/server/issues/12249

> Another thing which confuses me is that I can enforce two factor auth even if no single two factor out method is enabled. I think it would make sense to check that at least one is enabled (u2f, totp,...) and otherwise disable the checkbox.

**Comment by ChristophWurst** [comment]
Yep, that is what I also did at https://github.com/nextcloud/server/blob/8177fdb0f67a7fdfc86c27b3995afd9e5adfdce8/lib/private/Authentication/TwoFactorAuth/Manager.php#L118. This one is a special provider and it's fine to have an exception for it.

**Comment by szaimen** [comment]
It seems like this is still valid.

**Comment by i2h3** [comment]
This is still relevant and valid. Especially for service providers of managed Nextcloud deployments which do not offer CLI access to their end users, thus leaving the latter unable to help themselves in case they locked themselves out.

## Steps To Reproduce

1. Deploy a Nextcloud Docker container (TOTP app is disabled by default).
2. Go to administrative "Security" settings.
3. Enable "Enforce two-factor authentication".
4. Select "admin" in the "Enforced groups".
5. Save changes.
6. Log out.
7. Attempt to log in again (which no longer is possible).
